pyuvdata/uvflag/uvflag.py

```diff
@@ -1550,16 +1550,13 @@ def __add__(
             )
             this.Npols = len(this.polarization_array)
 
-        if this.mode == "flag":
-            this.flag_array = np.concatenate(
-                [this.flag_array, other.flag_array], axis=ax
-            )
-        elif this.mode == "metric":
-            this.metric_array = np.concatenate(
-                [this.metric_array, other.metric_array], axis=ax
-            )
-            this.weights_array = np.concatenate(
-                [this.weights_array, other.weights_array], axis=ax
+        # Will fail with AttributeError if the objects have different _data_params
+        # Might want a more straight-forward check ahead of time
```

Comment:
This comment makes me a little nervous. I would prefer if the check was added as a part of this PR. We already know they must be the same `mode` and `type` which, as long as the user hasn't altered the object's state, forces the required parameters to be the same. So this may not actually be an issue for required parameters.

The other question is what to do if the optional parameters exist on one and not the other? 
Erroring seems extreme since it is an optional parameter. We could throw a warning, and then fill in the missing data with zeros? That seems a little weird though too but not sure what the best solution is.

Reply:
I think filling in a `weights_square_array` with zeros will cause problems down the line in the application I have in mind. I guess if this attribute was vestigial after a type conversion (metric to flag), and you were adding flag objects, maybe you would not want an error. Could be a warning in one case and an error in the other? Of course, as additional optional parameters are developed, we may have to rethink this line of reasoning.

Reply:
Yeah it definitely gets complicated :thinking: I'm okay with the warning/error differential depending on cases. But we'll still need to eloquently handle when parameters exist on one object and not the other. Maybe it just makes them incompatible in the case where they are not vestigial?

Reply:
I think that's a good start. I don't know what the best check is. Not sure how you feel about try/except. One way would be

```
for attr in this._data_params:
    try:
        setattr(this, attr, np.concatenate(...))
    except AttributeError:
        vestigial = some_check_for_vestigial (mode=='flag' for now?)
        if vestigial:
            warning
        else:
            raise ValueError("UVFlag objects are not compatible for addition")
```

Reply:
I think that could be a good start. I'm not a huge fan of the try/except (despite it being the _pythonic_ way). But could just have an `if not hasattr()` instead.

Reply:
Error message could of course be more verbose

Reply:
Oh, yeah `if not hasattr` is definitely simpler/fewer lines/easier to test